config/config.go

```diff
@@ -67,12 +67,12 @@ type Config struct {
 	GoogleCredentials    string `name:"google_credentials" env:"GOOGLE_CREDENTIALS" auth:"google,sensitive"`
 
 	// Azure Resource Manager ID for Azure Databricks workspace, which is exhanged for a Host
-	AzureResourceID string `name:"azure_workspace_resource_id" env:"DATABRICKS_AZURE_RESOURCE_ID" auth:"azure"`
+	AzureResourceID string `name:"azure_workspace_resource_id" env:"DATABRICKS_AZURE_RESOURCE_ID" auth:"azure-cli,azure-msi" auth_group:"azure"`
 
-	AzureUseMSI       bool   `name:"azure_use_msi" env:"ARM_USE_MSI" auth:"azure"`
-	AzureClientSecret string `name:"azure_client_secret" env:"ARM_CLIENT_SECRET" auth:"azure,sensitive"`
-	AzureClientID     string `name:"azure_client_id" env:"ARM_CLIENT_ID" auth:"azure"`
-	AzureTenantID     string `name:"azure_tenant_id" env:"ARM_TENANT_ID" auth:"azure"`
+	AzureUseMSI       bool   `name:"azure_use_msi" env:"ARM_USE_MSI" auth:"azure-msi" auth_group:"azure"`
+	AzureClientSecret string `name:"azure_client_secret" env:"ARM_CLIENT_SECRET" auth:"azure-client-secret,sensitive" auth_group:"azure"`
+	AzureClientID     string `name:"azure_client_id" env:"ARM_CLIENT_ID" auth:"azure-client-secret,azure-msi" auth_group:"azure"`
+	AzureTenantID     string `name:"azure_tenant_id" env:"ARM_TENANT_ID" auth:"azure-cli,azure-client-secret" auth_group:"azure"`
```

Comment:
I change `auth` tag to be a specific type of authentication in order to be able to describe which properties are used / not used for specific auth method. But since some of attributes are can be in more than 1 auth `validate` call will fail if they are used. So for this I introduce `auth_group` attribute to make sure that `validate` behaviour stay the same.

Reply:
Ah, I see. Is the intent to make it match the resolved `auth_type`?

If so, then the Google ones also need to be updated. It may be easier to keep `auth` as is and use a new tag to annotate the auth types (e.g. `auth_types`).